### PR TITLE
Remove indexing states from NormalizedURI

### PR DESCRIPTION
**Pull request by gmethvin**
Remove the indexing states (INDEXED, FALLBACKED, UNSCRAPE_FALLBACK) and rely on only the sequence number to see if a page has been indexed.

Also remove the indexing failure states (INDEX_FAILED, FALLBACK_FAILED, UNSCRAPE_FALLBACK_FAILED). Replace them with a healthcheck error, since they usually indicate a serious problem and they rarely occur (there are none in the production DB now). In the future we can have a table with more detailed information on indexing errors.

### CODE REVIEW

File: shoebox/app/com/keepit/model/NormalizedURI.scala [hunk 1]
```diff
@@ -169,26 +169,14 @@ object NormalizedURIStates extends States[NormalizedURI] {
...
       (ACTIVE -> Set(SCRAPED, SCRAPE_FAILED, UNSCRAPABLE, INACTIVE)),
-      (SCRAPED -> Set(ACTIVE, INDEXED, INDEX_FAILED, INACTIVE)),
-      (SCRAPE_FAILED -> Set(ACTIVE, FALLBACKED, FALLBACK_FAILED, INACTIVE)),
-      (UNSCRAPABLE -> Set(ACTIVE, UNSCRAPE_FALLBACK, UNSCRAPE_FALLBACK_FAILED, INACTIVE)),
-      (INDEXED -> Set(ACTIVE, SCRAPED, INACTIVE, INDEXED)),
-      (INDEX_FAILED -> Set(ACTIVE, SCRAPED, INACTIVE, INDEX_FAILED)),
-      (FALLBACKED -> Set(ACTIVE, SCRAPE_FAILED, INACTIVE, FALLBACKED)),
-      (FALLBACK_FAILED -> Set(ACTIVE, SCRAPE_FAILED, INACTIVE, FALLBACK_FAILED)),
-      (UNSCRAPE_FALLBACK -> Set(ACTIVE, UNSCRAPABLE, INACTIVE, UNSCRAPE_FALLBACK)),
-      (UNSCRAPE_FALLBACK_FAILED -> Set(ACTIVE, UNSCRAPABLE, INACTIVE, UNSCRAPE_FALLBACK_FAILED)),
```
Comment (2is10) [inline shoebox/app/com/keepit/model/NormalizedURI.scala:191]: less state is always better state

File: shoebox/app/views/admin/bookmarks.scala.html [hunk 2]
```diff
@@ -85,3 +87,16 @@
   </table>
   @pagination(page, pageCount)
 }
+
+<script type="text/javascript">
+  $.get("@routes.ArticleIndexerController.getSequenceNumber", function (data) {
```
Comment (ymatsuda) [inline shoebox/app/views/admin/bookmarks.scala.html:92]: Is this sequence number always committed to the disk? It seems this seq number is the one indexer is working most recently, but not necessarily committed to the disk.

File: shoebox/app/com/keepit/model/ScrapeInfo.scala [hunk 3]
```diff
@@ -96,11 +96,9 @@ class ScrapeInfoRepoImpl @Inject() (val db: DataBaseComponent) extends DbRepo[Sc
   }
 
   def allActive(implicit session: RSession): Seq[ScrapeInfo] = {
-    val q = (for {
-       (s, u) <- table innerJoin inject[NormalizedURIRepoImpl].table on (_.uriId is _.id)
-       if u.state is NormalizedURIStates.INDEXED
-     } yield s.*)
-   q.list
+    (for {
+      (s, u) <- table innerJoin inject[NormalizedURIRepoImpl].table on (_.uriId is _.id)
```
Comment (andrewconner) [inline shoebox/app/com/keepit/model/ScrapeInfo.scala]: normalizedUriRepo can be provided in the constructor.

File: shoebox/app/views/admin/bookmarks.scala.html (hunk 2 above)
Comment (gmethvin) [inline shoebox/app/views/admin/bookmarks.scala.html:92]: It should be now. We're updating the value on every commit.

Comment (ymatsuda) [inline shoebox/app/views/admin/bookmarks.scala.html:92]: Oh. I see. Looks good.

**Pull request merged**